static/js/settings_emoji.js

```diff
@@ -183,6 +183,16 @@ exports.set_up = function () {
                 emoji[obj.name] = obj.value;
             }
 
+            if (emoji.name === "") {
+                ui_report.message(
+                    i18n.t("Failed: You must provide a name to the emoji."),
```

Comment:
I think we should use `ui_report.error(...)`, and I'd phrase it with proper English as "Emoji name is required."

Reply:
ui_report.error(...) requires an "xhr" parameter, which is undefined where I have written the check statement.

Reply:
@timabbott If no emoji name is present, then the post request sends a "404: Not Found" error, which sends HTML instead of a JSON response. So can't use ui_report.error() since it requires a JSON response(xhr) based on which it prints the error messages.